client/asset/eth/nodeclient.go

```diff
@@ -500,6 +500,32 @@ func (n *nodeClient) isRedeemable(secretHash [32]byte, secret [32]byte, contract
 	})
 }
 
+// transactionConfirmations gets the number of confirmations for the specified
+// transaction.
+func (n *nodeClient) transactionConfirmations(ctx context.Context, txHash common.Hash) (uint32, error) {
+	// We'll check the local tx pool first, since from what I can tell, a light
+	// client always requests tx data from the network for anything else.
+	if tx := n.leth.ApiBackend.GetPoolTransaction(txHash); tx != nil {
+		return 0, nil
+	}
+	hdr, err := n.bestHeader(ctx)
+	if err != nil {
+		return 0, err
+	}
+	tx, _, blockHeight, _, err := n.leth.ApiBackend.GetTransaction(ctx, txHash)
```

Comment:
OK, but it's curious that they do this first internally: https://github.com/ethereum/go-ethereum/blob/40cdcf8c47ff094775aca08fd5d94051f9cf1dbb/internal/ethapi/api.go#L1158-L1168
Could it be possible that there would be a result from `GetPoolTransaction` when `GetTransaction` would find it?
You got a CoinNotFoundError once...

Reply:
It makes more chronological sense to check the pool first, otherwise you might miss it checking "finalized" transactions first and it gets "finalized" before you check the pool where it was just removed.